src/main/java/com/azure/cosmos/examples/changefeed/SampleChangeFeedProcessor.java

```diff
@@ -0,0 +1,269 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.examples.changefeed;
+
+import com.azure.cosmos.ChangeFeedProcessor;
+import com.azure.cosmos.ConnectionPolicy;
+import com.azure.cosmos.ConsistencyLevel;
+import com.azure.cosmos.CosmosAsyncClient;
+import com.azure.cosmos.CosmosAsyncContainer;
+import com.azure.cosmos.CosmosAsyncDatabase;
+import com.azure.cosmos.CosmosClientBuilder;
+import com.azure.cosmos.CosmosClientException;
+import com.azure.cosmos.examples.common.CustomPOJO;
+import com.azure.cosmos.implementation.Utils;
+import com.azure.cosmos.models.CosmosAsyncContainerResponse;
+import com.azure.cosmos.models.CosmosContainerProperties;
+import com.azure.cosmos.models.CosmosContainerRequestOptions;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import reactor.core.scheduler.Schedulers;
+
+import java.time.Duration;
+import java.util.List;
+
+/**
+ * Sample for Change Feed Processor.
+ */
+public class SampleChangeFeedProcessor {
+
+    public static int WAIT_FOR_WORK = 60000;
+    public static final String DATABASE_NAME = "db_" + RandomStringUtils.randomAlphabetic(7);
+    public static final String COLLECTION_NAME = "coll_" + RandomStringUtils.randomAlphabetic(7);
+    private static final ObjectMapper OBJECT_MAPPER = Utils.getSimpleObjectMapper();
+    protected static Logger logger = LoggerFactory.getLogger(SampleChangeFeedProcessor.class.getSimpleName());
+
+
+    private static ChangeFeedProcessor changeFeedProcessorInstance;
+    private static boolean isWorkCompleted = false;
+
+    public static void main(String[] args) {
+        logger.info("BEGIN Sample");
+
+        try {
+            //This sample models an application where documents are being inserted into one container (the "feed container"),
+            //and meanwhile another worker thread or worker application is pulling inserted documents from the feed container's Change Feed
+            //and operating on them in some way. For one or more workers to process the Change Feed of a container, the workers must first contact the server
+            //and "lease" access to monitor one or more partitions of the feed container. The Change Feed Processor Library
```

Comment:
We should try to simplify this comment and add the following:

"A secondary Cosmos container ("lease container") is required in order to keep track of the documents being processed and resume the work from where it was left in case the processing is stopped. When the "feed container" has multiple partitions, corresponding number of "lease documents" are created and the Change Feed Processor instance will create equal amount of individual workers that will automatically process each partition in a multithread manner. In such cases the user's provided handler implementation can be invoked from different threads and execute in parallel. Multiple instances of the Change Feed Processor (i.e. running on different virtual machines) can operate on the same "feed container" at the same time to further parallelize the workload. Change Feed Processor library implement a load balancing component that will ensure more or less equal distribution of the workload among different instances."